File: frontend/src/pages/dashboard/teams/types/team.ts
export interface Team {
  id: number;
  name: string;
  description: string | null;
  company_id: number;
  department_id: number;
}

export interface TeamCreate {
  name: string;
  description?: string;
  company_id: number;
  department_id: number;
}

export interface TeamUpdate {
  name?: string;
  description?: string;
  department_id?: number;
}

File: backend/app/schemas/team.py
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class TeamBase(BaseModel):
    name: str = Field(..., min_length=2, max_length=100)
    description: Optional[str] = Field(default=None, max_length=255)
    company_id: int
    department_id: int


class TeamCreate(TeamBase):
    pass


class TeamUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=2, max_length=100)
    description: Optional[str] = Field(default=None, max_length=255)
    company_id: Optional[int] = None
    department_id: Optional[int] = None


class TeamResponse(TeamBase):
    id: int

    model_config = ConfigDict(from_attributes=True)